src/main/java/com/thebois/views/game/StructureButton.java

```diff
@@ -0,0 +1,87 @@
+package com.thebois.views.game;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Locale;
+
+import com.badlogic.gdx.scenes.scene2d.InputEvent;
+import com.badlogic.gdx.scenes.scene2d.ui.Button;
+import com.badlogic.gdx.scenes.scene2d.ui.Label;
+import com.badlogic.gdx.scenes.scene2d.ui.Skin;
+import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
+
+import com.thebois.listeners.IEventListener;
+import com.thebois.listeners.IEventSource;
+import com.thebois.listeners.events.OnClickEvent;
+import com.thebois.models.world.structures.StructureType;
+import com.thebois.utils.StringUtils;
+
+/**
+ * A button that represent a structure and is used to select which structure to build.
+ */
+public class StructureButton extends Button implements IEventSource<OnClickEvent<StructureType>> {
+
+    private static final float TOP_AND_BOTTOM_PADDING = 5f;
+    private static final float LEFT_AND_RIGHT_PADDING = 20f;
+    private final Collection<IEventListener<OnClickEvent<StructureType>>> onClickListeners =
+        new ArrayList<>();
+    private final StructureType structureType;
+    private final Button button;
+
+    /**
+     * Instantiate a structure button with the type of structure it should represent.
+     *
+     * @param structureType The type or structure to represent.
+     * @param skin          The skin used to display the button.
+     */
+    public StructureButton(final StructureType structureType, final Skin skin) {
+        super(skin, "toggle");
+        button = this;
```

Comment:
What are you doing here? I don't understand.

You can just use `this` directly, you don't need to store it in a field.

Reply:
Fair enough I just thought `button` made it slightly more readable, but maybe `this` is better, I can change it back to this.

Reply:
Ah, I see. The reason this confuses me a lot is because it looks like you're referring to some other button-object, when in fact you're referring to `this` button instance. Since the class *is* a button, saying `this.isToggled` should be easy enough to understand.

Reply:
I now remembered one more reason why I did the change use `button`, `this` in the lambda function further down doesn't work.
But I've made some changes, tell me what you think.